atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/AtlasRestoreService.java

```diff
@@ -163,15 +162,15 @@ public static AtlasRestoreService createForTests(
      *  Namespaces are repaired serially. If repairTable throws an exception, then this will propagate back to the
      *  caller. In such cases, some namespaces may not have been repaired.
      *
-     * @param namespaces the namespaces to repair.
+     * @param restoreRequests the repair requests.
      * @param repairTable supplied function which is expected to repair the given ranges.
      *
      * @return the set of namespaces for which we issued a repair command via the provided Consumer.
      */
     public Set<Namespace> repairInternalTables(
-            Set<Namespace> namespaces, BiConsumer<String, RangesForRepair> repairTable) {
-        Map<Namespace, CompletedBackup> completedBackups = getCompletedBackups(namespaces);
-        Set<Namespace> namespacesToRepair = completedBackups.keySet();
+            Set<RestoreRequest> restoreRequests, BiConsumer<String, RangesForRepair> repairTable) {
```

Comment:
We should probably check somewhere that we are not trying to restore two namespaces into the same one

Reply:
Adding a precondition in prepareRestore (and test).